go.mod

```diff
@@ -9,8 +9,7 @@ require (
 	github.com/adlio/schema v1.1.13
 	github.com/btcsuite/btcd v0.21.0-beta
 	github.com/btcsuite/btcutil v1.0.2
-	github.com/celestiaorg/nmt v0.10.0
-	github.com/celestiaorg/rsmt2d v0.6.0
```

Comment:
on re-review, it's not immediately clear why this dependency was removed in this PR. This repo doesn't depend on rsmt2d (see [search](https://github.com/celestiaorg/celestia-core/search?q=rsmt2d)) but nmt never depended on it so upgrading from v0.10.0 -> v0.11.0 didn't remove a transitive dependency.

I suspect, dependabot runs `go mod tidy` after updating a dependency. But I'm still curious as to how a previous commit that removed usage of rsmt2d didn't remove it.

Reply:
From https://github.com/celestiaorg/celestia-core/commit/79548a4f22d5124d8a8c37e507096d98698f8546

```
$ go mod why -m github.com/celestiaorg/rsmt2d
# github.com/celestiaorg/rsmt2d
(main module does not need module github.com/celestiaorg/rsmt2d)
```

From https://github.com/celestiaorg/celestia-core/commit/fcf45838c33f313016f0c3c4d5019b28618e8a46

```
$ go mod why -m github.com/celestiaorg/rsmt2d
# github.com/celestiaorg/rsmt2d
github.com/tendermint/tendermint/pkg/consts
github.com/celestiaorg/rsmt2d
```

Reply:
Note that I ran go mod tidy locally on a version before this PR here and it also removed rsmt2d.

Reply:
`go mod tidy` hasn't been run since https://github.com/celestiaorg/celestia-core/commit/97c3cce834c0fd0568bebf44c6c2fde6b06626e5

Reply:
we should have a CI check in this repo that fails if `go mod tidy` introduces any git diff